src/backend/joanie/tests/core/test_models_product.py

```diff
@@ -239,3 +240,48 @@ def test_model_product_get_equivalent_course_run_languages(self):
             product.get_equivalent_course_run_languages(),
             ["tr", "ast", "ne", "it", "af", "ro", "fr"],
         )
+
+    def test_model_product_get_equivalent_course_run_dates(self):
+        """
+        Check that product dates are processed
+        by aggregating target course runs dates as expected.
+        """
+        earliest_start_date = django_timezone.now() - timedelta(days=1)
+        latest_end_date = django_timezone.now() + timedelta(days=2)
+        latest_enrollment_start_date = django_timezone.now() - timedelta(days=2)
+        earliest_enrollment_end_date = django_timezone.now() + timedelta(days=1)
+
+        courses = (
+            factories.CourseRunFactory(
+                start=earliest_start_date,
+                end=latest_end_date,
+                enrollment_start=latest_enrollment_start_date - timedelta(days=1),
+                enrollment_end=earliest_enrollment_end_date + timedelta(days=1),
+            ).course,
+            factories.CourseRunFactory(
+                start=earliest_start_date + timedelta(days=1),
+                end=latest_end_date - timedelta(days=1),
+                enrollment_start=latest_enrollment_start_date,
+                enrollment_end=earliest_enrollment_end_date,
+            ).course,
+        )
+        product = factories.ProductFactory(target_courses=courses)
+
+        self.assertEqual(
+            product.get_equivalent_course_run_dates(),
+            {
+                "start": earliest_start_date,
+                "end": latest_end_date,
+                "enrollment_start": latest_enrollment_start_date,
+                "enrollment_end": earliest_enrollment_end_date,
+            },
+        )
```

Comment:
you should also test the result of product.state as nothing guarantees it is calling the `get_equivalent_course_run_dates` method?

Reply:
There is a test to check that product.state produce an expected result.